doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,812 @@
+
+```c# (but really MIDL3)
+namespace Microsoft.Management.Deployment
+{
+    /// State of the install.
+    enum InstallProgressState
+    {
+        /// The install is queued but not yet active. Cancellation of the IAsyncOperationWithProgress in this state will prevent the app from downloading or installing.
+        Queued = 0,
+        /// The installer is downloading. Cancellation of the IAsyncOperationWithProgress in this state will end the download and prevent the app from installing.
+        Downloading = 1,
+        /// The install is in progress. Cancellation of the IAsyncOperationWithProgress in this state will not stop the installation or the post install cleanup.
+        Installing = 2,
+        /// The installer has completed and cleanup actions are in progress. Cancellation of the IAsyncOperationWithProgress in this state will not stop cleanup or roll back the install.
+        PostInstall = 3,
+    };
+
+    /// Progress object for the install
+    /// DESIGN NOTE: percentage for the install as a whole is purposefully not included as there is no way to estimate progress when the installer is running.
+    [version(1)]
+    struct InstallProgress
+    {
+        /// State of the install
+        InstallProgressState State;
+        /// DESIGN NOTE: BytesDownloaded may only be available for downloads done by Windows Package Manager itself.
+        /// Number of bytes downloaded if known
+        UInt32 BytesDownloaded;
+        /// DESIGN NOTE: BytesRequired may only be available for downloads done by Windows Package Manager itself.
+        /// Number of bytes required if known
+        UInt32 BytesRequired;
+        /// Download percentage completed
+        UInt32 DownloadPercentage;
+    };
+
+    /// Result of the install
+    runtimeclass InstallResult
+    {
+        InstallResult();
+
+        /// Used by a caller to correlate the install with a caller's data.
+        String CorrelationId{ get; };
+        /// Whether a restart is required to complete the install.
+        Boolean RebootRequired { get; };
+    }
+
+    /// IMPLEMENTATION NOTE: SourceOrigin from AppInstallerRepositorySource.h
+    /// Defines the origin of the app catalog details.
+    enum AppCatalogOrigin
+    {
+        Default,
+        /// User means it was added by the user and could be removed.
+        User,
+        /// Predefined means it came as part of the Windows Package Manager package and cannot be removed.
+        Predefined,
+    };
+
+    /// IMPLEMENTATION NOTE: SourceTrustLevel from AppInstallerRepositorySource.h
+    /// Defines the trust level of the app catalog.
+    enum AppCatalogTrustLevel
+    {
+        None,
+        Trusted,
+    };
+
+    /// IMPLEMENTATION NOTE: SourceDetails from AppInstallerRepositorySource.h
+    /// Interface for retrieving information about an app catalog without acting on it.
+    [version(1)]
+    struct AppCatalogDetails
+    {
+        /// The name of the app catalog.
+        String Name;

```

Comment:
Is this a display name, like something that would be localized? Or is it pre-localized?

Reply:
Unlocalized. The existing built in one is:
PS C:\> winget source list
Name   Argument
-----------------------------------------
winget https://winget.azureedge.net/cache

So name is "winget". If someone added a source it would be whatever name they used when they used the command line. "winget source add --name Contoso https://www.contoso.com/cache" would end up with this API returning name "Contoso" for that catalog.